src/components/application_manager/src/commands/mobile/system_request.cc

```diff
@@ -53,12 +56,360 @@ uint32_t SystemRequest::index = 0;
 const std::string kSYNC = "SYNC";
 const std::string kIVSU = "IVSU";
 
+static const char* kQeeryAppsValidationFailedPrefix =
+    ":QUERY_APPS_VALIDATION_FAILED: ";
+
+template <typename ObjectType>
+class QueryAppsDataValidatorTemplate {
+ public:
+  typedef std::set<std::string> SynonymsSetType;
+  typedef std::map<std::string, SynonymsSetType> SynonymsMapType;
+
+  QueryAppsDataValidatorTemplate(const ObjectType& object)
+      : data_(object)
+      , error_prefix_(kQeeryAppsValidationFailedPrefix)
+      , max_vr_synonym_length_(40)
+      , min_vr_synonym_length_(1)
+      , max_tts_name_length_(500)
+      , max_vr_array_size_(100)
+      , min_vr_array_size_(1)
+      , max_url_scheme_lengh_(255)
+      , max_package_name_length_(255)
+      , max_app_id_length_(40)
+      , max_app_name_length_(100)
+      , max_languages_array_size_(100) {}
+
+  bool Validate() const {
+    if (!data_.isValid()) {
+      LOG4CXX_ERROR(logger_,
+                    error_prefix_ << "QueryApps response is not valid.");
+      return false;
+    }
+    if (!HasResponseKey()) {
+      return false;
+    }
+    bool result = ValidateAppDataAndOsAndLanguagesData();
+    app_ids_map_.clear();
+    synonyms_map_.clear();
+
+    return result;
+  }
+
+ private:
+  bool HasResponseKey() const {
+    if (!data_.keyExists(json::response)) {
+      LOG4CXX_ERROR(logger_,
+                    error_prefix_ << "QueryApps response does not contain '"
+                                  << json::response << "' parameter.");
+      return false;
+    }
+    return true;
+  }
+
+  bool ValidateAppDataAndOsAndLanguagesData() const {
+    const smart_objects::SmartArray* obj_array =
+        data_[json::response].asArray();
+    if (NULL == obj_array) {
+      LOG4CXX_ERROR(logger_,
+                    error_prefix_ << "QueryApps response is not array.");
+      return false;
+    }
+    bool result = true;
+    const std::size_t arr_size(obj_array->size());
```

Comment:
it's a valid way to initialize an integral values. when it's used without parameters it guaranties that variables get default values, for example int i(); guaranties that i will be initialized by 0 value.